src/Features/Core/Portable/Completion/Providers/TypeImportCompletionService.cs

```diff
@@ -314,17 +345,123 @@ public TypeOverloadInfo Aggregate(INamedTypeSymbol type)
+                        _isCSharp,
+                        _commonItemsBuilder.ToImmutableAndFree(),
+                        _genericItemsBuilder.ToImmutableAndFree(),
+                        _attributeWithSuffixItemsBuilder.ToImmutableAndFree());
+                }
+
+                public void AddItem(INamedTypeSymbol symbol, string containingNamespace, bool isPublic)
+                {
+                    ArrayBuilder<TypeImportCompletionItemInfo> correspondingBuilder;
+
+                    // Attribute type can't be generic
+                    if (symbol.Arity > 0)
+                    {
+                        correspondingBuilder = _genericItemsBuilder;
+                    }
+                    else if (IsAttributeWithAttributeSuffix(symbol))
+                    {
+                        correspondingBuilder = _attributeWithSuffixItemsBuilder;
+                    }
+                    else
+                    {
+                        correspondingBuilder = _commonItemsBuilder;
+                    }
+
+                    var item = TypeImportCompletionItem.Create(symbol, containingNamespace, _isCSharp);
+                    item.IsCached = true;
+                    correspondingBuilder.Add(new TypeImportCompletionItemInfo(item, isPublic));
+
+                    static bool IsAttributeWithAttributeSuffix(INamedTypeSymbol symbol)
+                    {
+                        // Do the symbol textual check first. Then the more expensive symbolic check.
+                        return symbol.Name.HasAttributeSuffix(isCaseSensitive: false) && symbol.IsAttribute();
+                    }
+                }
+            }
+
+            private ReferenceCacheEntry(
                 Checksum checksum,
-                ImmutableArray<TypeImportCompletionItemInfo> cachedItems)
+                bool isCSharp,
+                ImmutableArray<TypeImportCompletionItemInfo> commonItems,
+                ImmutableArray<TypeImportCompletionItemInfo> genericItems,
+                ImmutableArray<TypeImportCompletionItemInfo> attributeItems)
             {
                 Checksum = checksum;
-                CachedItems = cachedItems;
+                IsCSharp = isCSharp;
+
+                CommonItems = commonItems;
+                AttributeItems = attributeItems;
+                GenericItems = genericItems;
             }
 
+            public bool IsCSharp { get; }
+
             public Checksum Checksum { get; }
 
-            public ImmutableArray<TypeImportCompletionItemInfo> CachedItems { get; }
+            public ImmutableArray<TypeImportCompletionItemInfo> CommonItems { get; }
+
+            private ImmutableArray<TypeImportCompletionItemInfo> GenericItems { get; }
+
+            // Attribute types can't be generic
+            private ImmutableArray<TypeImportCompletionItemInfo> AttributeItems { get; }
+
+            public ImmutableArray<TypeImportCompletionItemInfo> GetGenericItems(bool isCSharp)
```

Comment:
>bool isCSharp [](start = 80, length = 13)

Can we please avoid this?
I do not like language specific code in /Core/.
If we have to use it, bool seems bad for API in the case.

Reply:
Fixed